lib/SIL/Parser/ParseSIL.cpp

```diff
@@ -121,7 +121,14 @@ ParseSILModuleRequest::evaluate(Evaluator &evaluator,
   SILParserState parserState(silMod.get());
   Parser parser(*bufferID, *SF, parserState.Impl.get());
   PrettyStackTraceParser StackTrace(parser);
-  parser.parseTopLevelSIL();
+
+  auto hadError = parser.parseTopLevelSIL();
+  if (hadError) {
+    // The rest of the SIL pipeline expects well-formed SIL, so if we encounter
+    // a parsing error, just return an empty SIL module.
+    return SILModule::createEmptyModule(mod, desc.conv, desc.opts,
+                                        desc.isWholeModule());
```

Comment:
Why not just return the incomplete silMod you already started parsing?

Reply:
@slavapestov The rest of the SIL pipeline cannot handle the invalid SIL it might contain, e.g not being in SSA form, type mismatches etc. We could probably recover better than just throwing everything away, but given SIL parsing is currently mostly used for testing, I'm not sure how much benefit we'd get from recovery.